tests/test_zogy.py

```diff
@@ -44,6 +44,171 @@ def setup_module(module):
     lsst.utils.tests.init()
 
 
+def makeFakeImages(size=(256, 256), svar=0.04, tvar=0.04, psf1=3.3, psf2=2.2, offset=None,
+                   psf_yvary_factor=0., varSourceChange=1/50., theta1=0., theta2=0.,
+                   n_sources=50, seed=66, verbose=False):
+    """Make two exposures: science and template pair with flux sources and random noise.
+    In all cases below, index (1) is the science image, and (2) is the template.
+
+    Parameters
+    ----------
+    size : `tuple` of `int`
+        Image pixel size (x,y). Pixel coordinates are set to
+        (-size[0]//2:size[0]//2, -size[1]//2:size[1]//2)
+    svar, tvar : `float`, optional
+        Per pixel variance of the added noise.
+    psf1, psf2 : `float`, optional
+        std. dev. of (Gaussian) PSFs for the two images in x,y direction. Default is
+        [3.3, 3.3] and [2.2, 2.2] for im1 and im2 respectively.
+    offset : `float`, optional
+        add a constant (pixel) astrometric offset between the two images.
+    psf_yvary_factor : `float`, optional
+        psf_yvary_factor vary the y-width of the PSF across the x-axis of the science image (zero,
+        the default, means no variation)
+    varSourceChange : `float`, optional
+        varSourceChange add this amount of fractional flux to a single source closest to
+        the center of the science image.
+    theta1, theta2: `float`, optional
+        PSF Gaussian rotation angles in degrees.
+    n_sources : `int`, optional
+        The number of sources to add to the images. If zero, no sources are
+        generated just background noise.
+    seed : `int`, optional
+        Random number generator seed.
+    verbose : `bool`, optional
+        Print some actual values.
+
+    Returns
+    -------
+    im1, im2 : `lsst.afw.image.Exposure`
+        The science and template exposures.
+
+    Notes
+    -----
+    If ``n_sources > 0`` and ``varSourceChange > 0.`` exactly one source,
+    that is closest to the center, will have different fluxes in the two
+    generated images. The flux on the science image will be higher by
+    ``varSourceChange`` fraction.
+
+    Having sources near the edges really messes up the
+    fitting (probably because of the convolution). So we make sure no
+    sources are near the edge.
+
+    Also it seems that having the variable source with a large
+    flux increase also messes up the fitting (seems to lead to
+    overfitting -- perhaps to the source itself). This might be fixed by
+    adding more constant sources.
+    """
+    rng = np.random.default_rng(seed)
+
+    psf1 = [3.3, 3.3] if psf1 is None else psf1
+    if not hasattr(psf1, "__len__") and not isinstance(psf1, str):
+        psf1 = [psf1, psf1]
+    psf2 = [2.2, 2.2] if psf2 is None else psf2
+    if not hasattr(psf2, "__len__") and not isinstance(psf2, str):
```

Comment:
This seems to be a relic from some previous version of the code, because it won't work if psf1 or psf2 are `str`. Maybe just remove the `and not isinstance(psf2, str)`?